apps/msgwakefup/boot.js

```diff
@@ -0,0 +1,5 @@
+// If doing regular loads, not Bangle.load, this is used:
+if (global.__FILE__=="messagegui.new.js") Bangle.setOptions({wakeOnFaceUp:true});
+
+// If Fastload Utils is installed this is used:
+Bangle.on("message", (_, msg)=>{if (Bangle.CLOCK && msg.new) {Bangle.setOptions({wakeOnFaceUp:true});}});
```

Comment:
Out of interest - do we get the intended functionality if we have just this `.on("message", ...)` code?

Reply:
I don't think so. In that case the the option would be set, then the bangle reset (thus setting the option to default value again) before loading messagesgui (and here the `__FILE__` logic sets the face up option again).

(reference https://github.com/espruino/BangleApps/pull/3334#issuecomment-2044596680)

Reply:
My understanding is that on fastload we don't run through the boot code, so the `__FILE__` check will not run since it only runs one time on boot.

Reply:
Gotya - out of interest, if you use `prependListener` here, does that help? Then the events will be:
- This handler, which sees there's a `.new` `"message"` and activates `wakeOnFaceUp`
- The message handler, which does whatever (vibrate the watch, alert the user)
- The user then twists their wrist and the screen wakes?

Whereas previously perhaps the message handler ran first and stopped this code from executing?

Reply:
You may be right - I hadn't thought of that. I think generally the option gets set when I expect it to. But maybe I've noticed some time that it didn't.

Reply:
Strange - let me know if you'd like a hand testing/investigating